test/test-ParametersParser.cc

```diff
@@ -84,7 +84,7 @@ TEST_CASE("Parse illegal parameter among legal ones", "[parameters]")
     "+ Parameters\n"\
     "    + OK_1\n"\
     "    + i;legal\n"\
-    "    + OK-2\n";
+    "    + OK_2\n";
```

Comment:
It changes the behaviour of old syntax only if the parameter has it's name and nothing else and that too there's a `-` in the name. Which is very rare and I think it should still parsed as MSON since `-` should expressed in URI encoded scheme.